daemon/src/state.rs

```diff
@@ -508,19 +515,24 @@ where
     S: Clone + Signer,
 {
     let pk = keys::PublicKey::from(peer.signer().public_key());
-    peer.using_storage(move |store| {
-        let malkovich = person::create(
-            store,
-            payload::Person {
-                name: Cstring::from(name),
-            },
-            Some(pk).into_iter().collect(),
-        )?;
+    let identity = peer
+        .using_storage(move |store| {
+            let malkovich = person::create(
+                store,
+                payload::Person {
+                    name: Cstring::from(name),
+                },
+                Some(pk).into_iter().collect(),
+            )?;
 
-        Ok::<_, Error>(local::load(store, malkovich.urn())?)
-    })
-    .await??
-    .ok_or(Error::IdentityCreationFailed)
+            Ok::<_, Error>(local::load(store, malkovich.urn())?)
+        })
+        .await??
+        .ok_or(Error::IdentityCreationFailed)?;
+
+    gossip::announce(peer, &identity.urn(), None);
```

Comment:
I don't think this needed, since the `announcement` loop will pick it up :)

Reply:
Addition of these 3 `announce`s is the reason why this PR exists in the first place :laughing: Without them the nodes wouldn't exchange information about person updates.

Reply:
Wuuuuut?! Even with the suggestion of changing the announcements code?

Reply:
This PR doesn't change the announcements code (probably yet). It adds 3 `announce`s, which are fixing the root issue and expands `build`, which IIUC should make the node listen for more announcements than before. I'm not even sure if we need to modify `build` in the first place, the issue is clearly about sending, not receiving announcements, the nodes were communicating fine with unmodified `build`.

Reply:
But it does change the `announcements`: https://github.com/radicle-dev/radicle-link/pull/689/commits/d11669af5bd40da6244a90d8ecc977607bcf622c

> node listen for more announcements than before

Other way around, it expands the announcements _made_ from just projects to projects & persons. This was suggested so we could remove adding `announce` to each `init`/`update` action, separating them from the annoucnement logic.

Reply:
Oh, I didn't notice that about `build` :facepalm:

I've created an experimental branch with `build` changes from this PR, but no explicit `announce` additions. Unfortunately it doesn't work well, upstream based on it apparently doesn't announce its person updates, they aren't visible on the other instance.

Reply:
> Oh, I didn't notice that about build 

Ahaha no worries :)

> Unfortunately it doesn't work well, upstream based on it apparently doesn't announce its person updates, they aren't visible on the other instance.

Huh, could you possibly run with logs enable, i.e. `RUST_LOG=librad::net=trace`, and see if you can spot announcements for `Person` or if they don't show up at all. Maybe sprinkle some logs for the announcements as well.

Reply:
Hmm, I can't spot announcements in logs in any version, with or without manual `announce`s. The worst part is that person changes don't seem to be propagated anymore, the other upstream node doesn't notice the updates :grimacing:

Reply:
Is there a specific Upstream branch you're working on that I can run and inspect?

Reply:
Yes, thank you!

I've backported changes from this PR to a commit used by upstream:
- With changes to `build`: [igor/announce-2](https://github.com/radicle-dev/radicle-link/tree/igor/announce-2)
- With changes to `build` and additional `announce`s: [igor/announce-3](https://github.com/radicle-dev/radicle-link/tree/igor/announce-3)

These backports are used in upstream:
- With changes to `build`: [igor/org-list-no-announce](https://github.com/radicle-dev/radicle-upstream/tree/igor/org-list-no-announce)
- With changes to `build` and additional `announce`s [igor/org-list-with-announce](https://github.com/radicle-dev/radicle-upstream/tree/igor/org-list-with-announce)

I've been trying to provoke person update announcements by linking the wallet to different addresses. The update to the document is made BEFORE the Ethereum transaction is sent. If you need any help, please reach out, it's not that trivial to get all the pieces together.

Reply:
Any news, @FintanH?

Reply:
Before I go down a rabbit hole, could you make an additional test similar to https://github.com/radicle-dev/radicle-link/blob/master/test/src/test/integration/daemon/gossip.rs#L19-L65 so that we test closer to the source?

Reply:
I've done some rebasing, there were some changes to the base upstream branch. Now it uses a newer version of link, which includes `can_observe_announcement_from_connected_peer`. It passes.

The link branches used in upstream:
- With changes to `build`: [announce-backport-no-explicit](https://github.com/radicle-dev/radicle-link/tree/igor/announce-backport-no-explicit)
- With changes to `build` and additional `announce`s: [igor/announce-backport](https://github.com/radicle-dev/radicle-link/tree/igor/announce-backport)

They are used in upstream:
- With changes to `build`: [igor/org-list-no-announce](https://github.com/radicle-dev/radicle-upstream/tree/igor/org-list-no-announce)
- With changes to `build` and additional `announce`s [igor/org-list-announce](https://github.com/radicle-dev/radicle-upstream/tree/igor/org-list-announce)

Reply:
> I've done some rebasing, there were some changes to the base upstream branch. Now it uses a newer version of link, which includes can_observe_announcement_from_connected_peer. It passes.

But can we add a test (or add to that test) a case where we have a `Person` update and witness the announcement? I believe that test only checks for a `Project`'s announcement.

Reply:
I've added the test. It fails before `igor/announce-backport-no-explicit` and passes since it. Explicit `announce`s don't make a difference.

Reply:
Ok, so that's a good sign that we at least know the announcement loop is working correctly. So we need to understand why this isn't happening in the Upstream environment, one oddity being that the explicit `announce` calls worked. 

On the same branch, were you able to see projects being announced and replicated?

Reply:
Wanted to highlight radicle-dev/radicle-upstream#1956 as it indicates broader issues with gossip in the context of Upstream.